src/newgidmap.c

```diff
@@ -196,6 +196,12 @@ int main(int argc, char **argv)
 		return EXIT_FAILURE;
 	}
 
+	/* Validate fds coming from get_pidfd_from_fd() */
```

Comment:
Would it, however, be better to do this in the get_pidfd_from_fd() fn itself?  I realize you have the struct stat here already, but any other callers of get_pidfd_from_fd() will need the same guarantee.

Reply:
I agree. Should I do the change?

Reply:
Yes, thanks.